.github/workflows/shellcheck.yml

```diff
@@ -0,0 +1,15 @@
+name: 'Shellcheck'
+
+on: [push,pull_request]
+
+jobs:
+  shellcheck:
```

Comment:
As you wish. This way it runs in parallel and the feedback is more immediate because if it fails you don't have to dig in the logs to discover that was the shellcheck.